### PR TITLE
feat: Bb 105 create a reusable footer component

### PR DESCRIPTION
**Pull request by GDamaso**
# Description

Created our footer component with BC Design Tokens and emotion styled components

## Type of changes

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

Desktop view VS. Mockup when running with Docker Compose (With older Dockerfile for hot-reloading)

Mobile View VS. Mockup:

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-126-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-126-frontend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

### CODE REVIEW

**Review by kcaparas1630 — CHANGES_REQUESTED** [review]
On Mobile View, assuming width 320, this is what I get. 
image
We should make it mobile-first then work on desktop next.

File: frontend/dev-dist/registerSW.js
Comment (LocalNewsTV) [inline frontend/dev-dist/registerSW.js:1]: Everything included in `dev-dist` should be removed, this is now being `.gitignored` 
The files are generative from running in dev and aren't needed

File: frontend/src/Views/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+const Footer = () => (
+  <StyledFooter>
+    <p>
+      By continuing, you agree to our&nbsp;
+      <a href="https://nmp.apps.nrs.gov.bc.ca/Information/Disclaimer">Terms of Service</a>
+    </p>
+    <p>
+      Need&nbsp;
+      <a href="https://www2.gov.bc.ca/gov/content/industry/agriculture-seafood/agricultural-land-and-environment/soil-nutrients/nutrient-management/nutrient-management-calculator/support-for-the-bc-nutrient-management-calculator">
```
Comment (LocalNewsTV) [inline frontend/src/Views/Footer.tsx:15]: Suggestion: Might be valuable to move your URLS into variables held in the component
1. Simplifies the `tsx`
2. Adds intent to what the urls are for (These urls are pretty self-explanatory though)

**Comment by LocalNewsTV** [comment]
I mentioned this before on the `Header` ticket 
Please include screenshots of mockups and your components in your PR :)
It helps me review you more fairly, and not operate on the assumptions of what things are meant to look like. 
Its also a good thing to get used to it, since it makes PRs far more descriptive.

**Comment by GDamaso** [comment]
@DerekRoberts  Backend is failing out of nowhere here. Any insights?

```bash
Run # Expand for route verification
  # Expand for route verification
  
  # Check for URL (route + path)
  URL_HOST_PATH=nr-sustainment-capstone-2024-126-backend.apps.silver.devops.gov.bc.ca/api/health
  if [ -z "${URL_HOST_PATH}" ]; then
    echo "No route found. Skipping."
    exit 0
  fi
  echo -e "URL: ${URL_HOST_PATH}"
  
  # Curl and verify
  TRIES="3"
  SECONDS="15"
  for (( i=0; i<"${TRIES}"; i++ )); do
    HTTP_CODE=$(curl -Lso /dev/null -w "%{http_code}" "${URL_HOST_PATH}")
    if [ "${HTTP_CODE}" -eq 200 ]; then
      echo -e "Route verification successful!\n"
      exit 0
    fi
    echo -e "HTTP_CODE:${HTTP_CODE}, Try: #${i}"
    sleep "${SECONDS}"
  done
  echo -e "\nRoute verification failed"
  exit 1
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
##[debug]/usr/bin/bash --noprofile --norc -e -o pipefail /home/runner/work/_temp/9210e767-6a5e-4ee5-8ae8-e59e473e1ff7.sh
URL: nr-sustainment-capstone-2024-126-backend.apps.silver.devops.gov.bc.ca/api/health
HTTP_CODE:503, Try: #0
HTTP_CODE:503, Try: #1
HTTP_CODE:503, Try: #2

Route verification failed
Error: Process completed with exit code 1.
```

No change in the backend has been done in a while and the header component PR doesn't seem to be deployed to our [openshift domain](https://nr-sustainment-capstone-2024-126-frontend.apps.silver.devops.gov.bc.ca/):



But a part of this PR seems to have been deployed, since there is a footer component over there.

Thanks in advance :smile:

**Comment by DerekRoberts** [comment]
@GDamaso Good troubleshooting!  :)

I'm off this week and really appreciate your efforts.  Please pull the most recent change, which bumps verification retries from 3 to 5, which can't hurt.

Backend is still failing, but not during deployment.  Are you okay with working on this and we team up Monday there's anything left?
```
node:internal/process/promises:289
triggerUncaughtException(err, true /* fromPromise */);
```
https://console.apps.silver.devops.gov.bc.ca/k8s/ns/b4cd74-dev/pods/nr-sustainment-capstone-2024-126-backend-84bdd6f944-h987k/logs

**Comment by GDamaso** [comment]
@DerekRoberts I don't mind at all. 
Thanks for all the help you've put into our project already,  we really appreciate it!
Enjoy your week off :tada:

**Review by kcaparas1630 — APPROVED** [review]
LGTM 🚀

**Pull request merged**